dj/construction/build.py

```diff
@@ -339,47 +343,144 @@ def _build_select_ast(
     _build_tables_on_select(session, select, tables, build_criteria)
 
 
-def add_filters_dimensions_limit_to_query_ast(
+def _consolidate_tables(query: ast.Query):
```

Comment:
I am having a hard time understanding how any changes made inside this function make it outside. Is it through the `query` argument?

Reply:
a lot of key functions in `build.py` modify in-place. `add_filters_dimensions_limit_to_query_ast` and `_consolidate_tables` both modify in-place by acting on their `query` arguments